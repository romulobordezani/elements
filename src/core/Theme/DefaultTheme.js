import DefaultColors from '../Colors/index';

class DefaultTheme {
  constructor(theme) {
    this.colors = new DefaultColors(theme.colors);

    this.currentTheme = 'default';

    this.palette = {
      primary: {
        light: this.colors.light.A100,
        main: this.colors.light.A700,
        dark: this.colors.dark.A700,
        contrastText: this.colors.light[50],
      },
      secondary: {
        light: this.colors.dark.A100,
        main: this.colors.dark.A700,
        dark: this.colors.dark.A400,
        contrastText: this.colors.light[50],
      },
    };

    this.text = {
      primary: 'rgba(0, 0, 0, 0.87)', // The most important text.
      secondary: 'rgba(0, 0, 0, 0.54)', // Secondary text.
      disabled: 'rgba(0, 0, 0, 0.38)', // Disabled text have even lower visual prominence.
      hint: 'rgba(0, 0, 0, 0.38)', // Text hints.
    };

    // The color used on the line which divides different contents.
    this.divider = 'rgba(0, 0, 0, 0.12)';

    // The background colors used to style the surfaces.
    // Consistency between these values is important.
    this.background = {
      paper: '#FFFFFF',
      default: '#CCC',
    };

    // The colors used to style the action elements.
    this.action = {
      active: 'rgba(0, 0, 0, 0.54)', // The color of an active action like an icon button.
      hover: 'rgba(0, 0, 0, 0.08)', // The colors of an hovered action.
      hoverOpacity: 0.08,
      selected: 'rgba(0, 0, 0, 0.14)', // The color of a selected action.
      disabled: 'rgba(0, 0, 0, 0.26)', // The color of a disabled action.
      disabledBackground: 'rgba(0, 0, 0, 0.12)', // The background color of a disabled action.
    };
  }
}

export default DefaultTheme;
