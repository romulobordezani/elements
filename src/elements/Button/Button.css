.eButton{color:red;border:2px solid green}
