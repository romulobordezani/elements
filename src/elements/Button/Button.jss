const styleOverride = {
  eButton: {
    backgroundColor: props => props.theme.palette.primary.main,
    color: props => props.theme.palette.primary.contrastText,
  },
};

export default styleOverride;
