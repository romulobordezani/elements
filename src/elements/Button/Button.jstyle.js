const styleOverride = {
  eButton: {
    backgroundColor: props => props.theme.palette.primary.main,
    color: props => props.theme.palette.primary.darkContrastText,
  },
};

export default styleOverride;
