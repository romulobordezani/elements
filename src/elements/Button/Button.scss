.eButton{
  color: red;
  border: 2px solid rgba(0,0,0,0.2);
  border-radius: 3px;
  transition: background-color, max-width .3s ease-out;
  outline: none;
  cursor: pointer;
  text-shadow: 1px 1px 1px #ADADAD;
}
