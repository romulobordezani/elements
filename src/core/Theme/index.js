import { merge } from '../../handlers/index';
import DefaultTheme from './DefaultTheme';

class Theme {
  constructor(theme) {
    let defaultTheme = new DefaultTheme(theme);
    defaultTheme = merge(defaultTheme, theme);
    Object.assign(this, defaultTheme);
  }
}

export default Theme;
