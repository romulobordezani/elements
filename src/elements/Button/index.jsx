import React from 'react';
import PropTypes from 'prop-types';
import Theme from '../../core/Theme/index';

const Button = function Button(props, context) {
  return (
    <button style={{ color: context.theme.palette.primary.dark }}>
      {props.content},
    </button>
  );
};

Button.propTypes = {
  content: PropTypes.string.isRequired,
};

Button.contextTypes = {
  theme: PropTypes.instanceOf(Theme).isRequired,
};

export default Button;
