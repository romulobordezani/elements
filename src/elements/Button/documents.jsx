// /* eslint-env browser */

import React from 'react';
import Markdown from 'react-markdown';
import MdCodeBlock from '../../viewer/components/ViewerContainer/MdCodeBlock';
import Button from './index';

const readme = require('./README.md');

const customButtonTricaeStyle = {
  margin: '10px',
  flexGrow: 1,
  boxShadow: '0 3px 0 0 #cd910a',
  borderRadius: '3px',
  backgroundColor: '#f4af16',
};

/* eslint-disable no-alert */
function ButtonDocs() {
  return (
    <div>
      <h2><a href="#button" className="anchor" id="button">Button</a></h2>
      <Markdown
        source={readme}
        escapeHtml={false}
        renderers={{ code: MdCodeBlock }}
      />
      Live example:
      <div style={{ display: 'flex', flexDirection: 'row', marginBottom: '60px' }} >
        <Button style={{ margin: '10px', flexGrow: 1 }} onClick={() => window.alert('I was clicked!')}>Default</Button>
        <Button style={{ margin: '10px', flexGrow: 1 }} palette="secondary" tone="dark">Secondary</Button>
        <Button style={customButtonTricaeStyle} palette="primary" tone="light">Tricae Bevel</Button>
        <Button style={{ margin: '10px', flexGrow: 1 }} palette="white" >White</Button>
        <Button style={{ margin: '10px', flexGrow: 1 }} palette="black" >Black</Button>
      </div>
    </div>
  );
}
/* eslint-enable no-alert */

export default ButtonDocs;
