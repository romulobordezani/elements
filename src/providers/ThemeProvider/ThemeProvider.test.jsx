import React from 'react';
import renderer from 'react-test-renderer';
import { ThemeContext } from './index';
import Theme from '../../core/Theme';
import Button from '../../elements/Button';

const theme = {
  theme: new Theme({}),
};

describe('Theme', () => {

  it('Renders a default theme, without any customization', () => {
    const tree = renderer.create(
      <ThemeContext.Provider value={theme}>
        <Button>Button</Button>
      </ThemeContext.Provider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Renders a CUSTOM theme, with new colors customized', () => {

    const customTheme = {
      theme: new Theme({
        palette: {
          primary: {
            dark: 'pink',
          },
          secondary: {
            light: 'blue',
          },
        },
      }),
    };

    const tree = renderer.create(
      <ThemeContext.Provider value={customTheme}>
        <Button>Button</Button>
      </ThemeContext.Provider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

});
