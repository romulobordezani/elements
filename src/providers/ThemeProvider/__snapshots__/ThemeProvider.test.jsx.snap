// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Theme Renders a CUSTOM theme, with new colors customized 1`] = `
<button
  className="eButton Button-eButton-0-1-3 Button-eButton-0-1-1"
>
  Button
</button>
`;

exports[`Theme Renders a default theme, without any customization 1`] = `
<button
  className="eButton Button-eButton-0-1-2 Button-eButton-0-1-1"
>
  Button
</button>
`;
